components/messaging/CurrentConversation.tsx

```diff
@@ -81,59 +83,67 @@ const CurrentConversation: React.FC = () => {
     };
   }, [currentMessages, setCurrentMessages]);
 
+  const debounceRef = useRef<ReturnType<typeof setTimeout> | null>(null);
+
   useEffect(() => {
     const handleScroll = () => {
       setIsScrolling(true);
 
-      const debounce = setTimeout(() => {
+      if (debounceRef.current) {
+        clearTimeout(debounceRef.current); // Clear the previous timeout
+      }
+
+      debounceRef.current = setTimeout(() => {
         setIsScrolling(false);
       }, 3000);
-
-      return () => clearTimeout(debounce);
     };
 
     chatWindowRef?.current?.addEventListener('scroll', handleScroll);
 
     return () => {
       chatWindowRef?.current?.removeEventListener('scroll', handleScroll);
+      if (debounceRef.current) {
+        clearTimeout(debounceRef.current); // Clear timeout on cleanup
```

Comment:
it looks like there are still some comments @AnyaVeselova

Reply:
Oh sorry! I'll delete it right now!